conans/model/build_info.py

```diff
@@ -145,7 +148,12 @@ def _filter_paths(self, paths):
     @property
     def build_modules_paths(self):
         if self._build_modules_paths is None:
-            if isinstance(self.build_modules, list):  # FIXME: This should be just a plain dict
+            # FIXME: This should be just a plain dict
+            if isinstance(self.build_modules, six.string_types):
+                conan_v2_behavior("Use 'self.cpp_info.build_modules[\"<generator>\"] = "
+                                  "{the_list}' instead".format(the_list=[self.build_modules, ]))
+                self.build_modules = BuildModulesDict.from_list([self.build_modules, ])
```

Comment:
Shall I open a different PR with this change? I'm not sure if we want to cover this scenario (maybe not), but the current error is very ugly... I preferred to implement it than to raise. wdyt?

Reply:
I would say raise always with a clear error message?

Reply:
Moving to a different PR

Reply:
No, I'm not suggesting this change anymore. It should be something to check and validate for _all_ `cpp_info` fields, it doesn't make sense to check just this one.